docs/gcloud-auth.rst

```diff
@@ -0,0 +1,264 @@
+
+=========================================
+Authenticating locally with the Cloud SDK
+=========================================
+
+This is the easiest way to authenticate while you're developing locally.
+
+#. Download and install the Cloud SDK (You can get the Cloud SDK at https://cloud.google.com/sdk)
+#. Authenticate using OAuth2 (``gcloud auth login``)
+#. Run your code (ie, ``python myscript.py``)
+
+
+After you do this,
+your script will look for the authentication token
+that you got when authenticating,
+and you won't need to think about authentication in your code at all.
+
+======================================
+Authenticating with a specific keyfile
+======================================
+
+----------------------------------------------
+Setting the keyfile in an environment variable
+----------------------------------------------
+
+If you have a specific JSON keyfile downloaded that you'd like to use,
+you can simply set the path to this in
+the ``GOOGLE_APPLICATION_CREDENTIALS`` environment variable.
+
+This means your code doesn't have to change at all,
+and can run with different credentials depending on the environment.
+
+Here's what this looks like:
+
+.. code-block:: bash
+
+   $ export GOOGLE_APPLICATION_CREDENTIALS="/path/to/key.json"
+   $ python
+   >>> from gcloud import pubsub
+   >>> pubsub.Topic('topic_name').create()
+
+--------------------------------------
+Setting the keyfile explicitly in code
+--------------------------------------
+
+If you really want to set a specific keyfile in code
+(maybe for a code snippet to send to a friend?)
+you can do this, but it's really not recommended.
+
+~~~~~~~~~~~~~~~~~~~~~~~~
+... using a JSON keyfile
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+   from gcloud.credentials import get_for_service_account_json
+   from gcloud import pubsub
+
+   # Create the credentials from the keyfile and set the default connection.
+   credentials = get_for_service_account_json('/path/to/key.json')
+   connection = pubsub.Connection(credentials=credentials)
+   pubsub.set_default_connection(connection)
+
+   # Now you can interact with the service as usual.
+   pubsub.Topic('topic_name').create()
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+... using a .p12 key and client email
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+   from gcloud.credentials import get_for_service_account_p12
+   from gcloud import pubsub
+
+   # Create the credentials from the .p12 file and set the default connection.
+   credentials = get_for_service_account_p12(
+       'special-email-for-p12@developer.gserviceaccount.com',
+       '/path/to/key.p12')
+   connection = pubsub.Connection(credentials=credentials)
+   pubsub.set_default_connection(connection)
+
+   # Now you can interact with the service as usual.
+   pubsub.Topic('topic_name').create()
+
+=====================================
+Authenticating from inside GCE or GAE
+=====================================
```

Comment:
Any issue to axe the declarative snippets I included in #805?

They are quite simple:

```python
from oauth2client.gce import AppAssertionCredentials
credentials = AppAssertionCredentials([])
# AND
from oauth2client.appengine import AppAssertionCredentials
credentials = AppAssertionCredentials([])
```